src/config/builder.rs

```diff
@@ -160,6 +161,7 @@ fn sort_json_value(value: &mut Value) {
 impl<'a> From<&'a ConfigBuilder> for ConfigBuilderHash<'a> {
     fn from(value: &'a ConfigBuilder) -> Self {
         ConfigBuilderHash {
+            version: env!("CARGO_PKG_VERSION"),
```

Comment:
I think we'd prefer to use `get_version()` here which is what we use to report a Vector version: https://github.com/vectordotdev/vector/blob/master/src/lib.rs#L121 This would then cover differences like `nightly` builds. cc: @leebenson if I'm missing context